addons/payment_transfer/models/payment_transaction.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+
+import logging
+import pprint
+
+from odoo import _, api, models
+from odoo.tools.float_utils import float_compare
+
+from odoo.addons.payment.models.payment_acquirer import ValidationError
+
+_logger = logging.getLogger(__name__)
+
+
+class TransferPaymentTransaction(models.Model):
+    _inherit = 'payment.transaction'
+
+    def _get_specific_rendering_values(self, processing_values):
+        self.ensure_one()
+        if self.provider != "transfer":
+            return super()._get_specific_rendering_values(processing_values)
+
+        return {
+            'currency': self.currency_id.name,
+            'tx_url': self.acquirer_id._transfer_get_redirect_action_url(),
+            **processing_values,
+        }
+
+    @api.model
+    def _get_tx_from_feedback_data(self, provider, data):
+        if provider != "transfer":
+            return super()._get_tx_from_feedback_data(provider, data)
+
+        reference = data.get('reference')
+        tx = self.search([('reference', '=', reference)])
+
+        if not tx or len(tx) > 1:
+            error_msg = _('received data for reference %s') % (pprint.pformat(reference))
+            if not tx:
+                error_msg += _('; no order found')
+            else:
+                error_msg += _('; multiple order found')
+            _logger.info(error_msg)
+            raise ValidationError(error_msg)
+
+        if float_compare(float(data.get('amount', '0.0')), tx.amount, 2) != 0:
+            raise ValidationError(_('Incorrect amount: received %.2f, expected %.2f', float(data.get('amount', '0.0')), tx.amount))
+        if data.get('currency') != tx.currency_id.name:
+            raise ValidationError(_('Incorrect currency: received %s, expected %s', data.get('currency'), tx.currency_id.name))
+
+        return tx
+
+    def _process_feedback_data(self, data):
+        if self.provider != "transfer":
+            return super()._process_feedback_data(data)
+
+        _logger.info('Validated transfer payment for tx %s: set as pending' % (self.reference))
+        self._set_pending()
+
+    def _get_sent_message(self):
+        self.ensure_one()
+        if self.provider == 'transfer':
+            return _(
+                "The customer has selected %(acq_name)s to make the payment.",
+                acq_name=self.acquirer_id.name
+            )
+
+        return super()._get_sent_message()
+
+    def _log_received_message(self):
+        self = self.filtered(lambda t: t.provider != 'transfer')
+        super()._log_received_message()
```

Comment:
Overwriting self is evil
```suggestion
        super(PaymentTransaction, self.filtered(lambda t: t.provider != 'transfer'))._log_received_message()
```